modules/nextflow/src/main/groovy/nextflow/processor/TaskArrayCollector.groovy

```diff
@@ -91,11 +91,13 @@ class TaskArrayCollector {
             // submit task directly if the collector is closed
             // or if the task is retried (since it might have dynamic resources)
             if( closed || task.config.getAttempt() > 1 ) {
+                task.isChild = false
```

Comment:
This is implicit because `boolean` is `false` by default

Reply:
I have explicitly set to false to ensure it is not treat as an array when task is retried.